crates/jstzd/src/task/inner_octez_node.rs

```diff
@@ -0,0 +1,79 @@
+#![allow(unused)]
+use std::{fs::File, path::PathBuf, process::Stdio};
+
+use tokio::process::{Child, Command};
+
+use anyhow::{anyhow, Result};
+
+fn path_or_default<'a>(path: Option<&'a PathBuf>, default: &'a str) -> &'a str {
+    path.and_then(|bin| bin.to_str()).unwrap_or(default)
+}
+
+async fn run_command(command: &mut Command) -> Result<()> {
+    let output = command.output().await?;
+
+    if !output.status.success() {
+        return Err(anyhow!(
+            "Command {:?} failed:\n {}",
+            command,
+            String::from_utf8_lossy(&output.stderr)
+        ));
+    }
+
+    Ok(())
+}
+
+pub struct InnerOctezNode {
+    /// Path to the octez-node binary
+    /// If None, the binary will inside PATH will be used
+    pub octez_node_bin: Option<PathBuf>,
+    /// Path to the octez-node directory
+    pub octez_node_dir: PathBuf,
+}
+
+impl InnerOctezNode {
+    fn command(&self) -> Command {
+        Command::new(path_or_default(self.octez_node_bin.as_ref(), "octez-node"))
+    }
+
+    pub async fn config_init(
+        &self,
+        network: &str,
+        http_endpoint: &str,
+        rpc_endpoint: &str,
+        num_connections: u32,
+    ) -> Result<()> {
+        run_command(self.command().args([
+            "config",
+            "init",
+            "--network",
+            network,
+            "--data-dir",
+            self.octez_node_dir.to_str().expect("Invalid path"),
+            "--net-addr",
+            http_endpoint,
```

Comment:
`--net-addr` controls the p2p connection point, not the http endpoint. Can be removed in favour of the default.